lxd/containers.go

```diff
@@ -356,7 +362,15 @@ func createFromImage(d *Daemon, req *containerPostReq) Response {
 			}
 
 			if !c.isPrivileged() {
-				return shiftRootfs(c, name, d)
+				err = shiftRootfs(c, name, d)
```

Comment:
One thing here is that it looks like a lot of stuff is duplicated in the btrfs cases everywhere. We should probably extract this somehow to keep it from getting out of hand when we add LVM as well.